crates/topos-tce-broadcast/src/double_echo/mod.rs

```diff
@@ -276,52 +507,79 @@ impl DoubleEcho {
     }
 
     #[cfg_attr(feature = "direct", allow(dead_code))]
-    pub(crate) fn handle_broadcast(&mut self, cert: Certificate) {
+    pub(crate) fn handle_broadcast(&mut self, cert: Certificate, origin: bool) {
```

Comment:
Not sure to get the purpose of this `origin`, I found it set only on the tests to true

Is it somewhat to track whether the certificate is coming from the API or from gossip or something else?

Reply:
Yep it's to define if the we need to gossip the certificate when receiving the cert via API